From 3a8595f1a57759044be301fc76a2300b7f2efe66 Mon Sep 17 00:00:00 2001
From: Benedikt Ritter <britter@apache.org>
Date: Tue, 21 Jan 2014 18:45:08 +0000
Subject: [PATCH] LANG-954: uncaught PatternSyntaxException in FastDateFormat -
 add test for the pattern in question; works at least with JDK 1.7.0_45

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/lang/trunk@1560137 13f79535-47bb-0310-9956-ffa450edef68
---
 .../apache/commons/lang3/time/FastDateFormatTest.java    | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/test/java/org/apache/commons/lang3/time/FastDateFormatTest.java b/src/test/java/org/apache/commons/lang3/time/FastDateFormatTest.java
index fce0c6efcd1..21a38f4b41c 100644
--- a/src/test/java/org/apache/commons/lang3/time/FastDateFormatTest.java
+++ b/src/test/java/org/apache/commons/lang3/time/FastDateFormatTest.java
@@ -250,6 +250,15 @@ public void testTimeDateDefaults() {
                 FastDateFormat.getDateTimeInstance(FastDateFormat.LONG, FastDateFormat.MEDIUM, TimeZone.getDefault(), Locale.getDefault()));
     }
 
+    /**
+     * According to LANG-954 (https://issues.apache.org/jira/browse/LANG-954) this is broken in Android 2.1.
+     */
+    @Test
+    public void testLang954() throws Exception {
+        String pattern = "yyyy-MM-dd'T'";
+        FastDateFormat.getInstance(pattern);
+    }
+
     @Test
     public void testParseSync() throws InterruptedException {
         final String pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS Z";
